### PR TITLE
Allow repeatOnLifecycle state to be configurable.

### PR DESCRIPTION
**Pull request by c-soon**
While most workflow use cases is likely to be inside an Activity and can specify renderings
to be updated while in the Lifecycle.State.STARTED state, there are certain use cases
where we want the workflow to depend on the application's lifecycle (using ProcessLifecycleOwner.)
which will not work when calling repeatOnLifecycle with Lifecycle.State.STARTED as this state event
is tied to Activities.

### CODE REVIEW

File: workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLayout.kt
Comment (rjrjr) [inline workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLayout.kt:76]:
```suggestion
   * @param [repeatOnLifecycle] the lifecycle state in which renderings should be actively. Defaults to STARTED, which is appropriate for Activity and Fragment.
```

**Review by rjrjr — APPROVED** [review]
You'll need to update the api files too.

```
./gradlew workflow-ui:core-android:apiDump
```

**Comment by rjrjr** [comment]
Thanks!

**Comment by c-soon** [comment]
> You'll need to update the api files too.
> 
> ```
> ./gradlew workflow-ui:core-android:apiDump
> ```

Oops, thank you! Updated!

File: workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLayout.kt
Comment (c-soon) [reply workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLayout.kt:76]: Updated.

Comment (c-soon) [reply workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLayout.kt:76]: updated!

**Comment by c-soon** [comment]
@rjrjr fyi this looks like its pending a maintainer to approve something? 

```First-time contributors need a maintainer to approve running workflows.```

**Comment by rjrjr** [comment]
> First-time contributors need a maintainer to approve running workflows.

That already happened, it's what's allowing CI to run.

**Comment by rjrjr** [comment]
The CI failures do look bogus, though. Trying again…

**Comment by c-soon** [comment]
Looks like it executed now, for some reason the 3 failed executions didn't run before, I'll address the failures i think i need to update the usages in the samples to named params

**Comment by rjrjr** [comment]
Oh, of course. Thanks.

**Comment by steve-the-edwards** [comment]
I approved them to run - thats why they started running. Sorry for the confusion!

**Comment by c-soon** [comment]
> I approved them to run - thats why they started running. Sorry for the confusion!

Ah, thank you! Does this need to be done every time I force push an update? I see the same message again.

**Comment by rjrjr** [comment]
> > I approved them to run - thats why they started running. Sorry for the confusion!
> 
> Ah, thank you! Does this need to be done every time I force push an update? I see the same message again.

That's weird, I think a recent development. @RBusarow, does this have anything to do with the new action to kill off previous runs? Any clue to share?

**Comment by RBusarow** [comment]
I believe this is a relatively new setting.

image

It's meant to prevent people/bots from PRing malicious actions and having them auto-execute.  I guess you technically don't count as a contributor until the first PR's merged.

**Comment by rjrjr** [comment]
> It's meant to prevent people/bots from PRing malicious actions and having them auto-execute. I guess you technically don't count as a contributor until the first PR's merged.

Gross. I guess it's been this way for a while, then. I never noticed before that I have to approve over and over again.

We had to turn this on after attracting some bitcoin creeps who messed with our actions. :/

**Pull request merged**
